leaderelection/config.go

```diff
@@ -0,0 +1,174 @@
+		if leaseDuration, err := time.ParseDuration(leaseDurationStr); err == nil {
+			config.LeaseDuration = leaseDuration
+		} else {
+			return nil, fmt.Errorf("leaseDuration: invalid duration: %q", leaseDurationStr)
+		}
+	} else {
+		return nil, errors.New("leaseDuration cannot be empty")
+	}
+
+	if renewDeadlineStr, ok := data["renewDeadline"]; ok {
+		if renewDeadline, err := time.ParseDuration(renewDeadlineStr); err == nil {
+			config.RenewDeadline = renewDeadline
+		} else {
+			return nil, fmt.Errorf("renewDeadline: invalid duration: %q", renewDeadlineStr)
+		}
+	} else {
+		return nil, errors.New("renewDeadline cannot be empty")
+	}
+
+	if retryPeriodStr, ok := data["retryPeriod"]; ok {
+		if retryPeriod, err := time.ParseDuration(retryPeriodStr); err == nil {
+			config.RetryPeriod = retryPeriod
+		} else {
+			return nil, fmt.Errorf("retryPeriod: invalid duration: %q", retryPeriodStr)
+		}
+	} else {
+		return nil, errors.New("retryPeriod cannot be empty")
+	}
+
+	// enabledComponents are not validated here, because they are dependent on
+	// the component. Components should provide additional validation for this
+	// field.
+	if enabledComponents, ok := data["enabledComponents"]; ok {
+		tokens := strings.Split(enabledComponents, ",")
+		config.EnabledComponents = sets.NewString(tokens...)
+	}
+
+	return config, nil
+}
+
+// NewConfigFromConfigMap returns a new Config from the given ConfigMap.
+func NewConfigFromConfigMap(configMap *corev1.ConfigMap) (*Config, error) {
+	if configMap == nil {
+		config := defaultConfig()
+		return &config, nil
+	}
+
+	return NewConfigFromMap(configMap.Data)
+}
+
+// Config represents the leader election config for a set of components
+// contained within a single namespace. Typically these will correspond to a
+// single source repository, viz: serving or eventing.
+type Config struct {
+	ResourceLock      string
+	LeaseDuration     time.Duration
+	RenewDeadline     time.Duration
+	RetryPeriod       time.Duration
+	EnabledComponents sets.String
+}
+
+func (c *Config) GetComponentConfig(name string) ComponentConfig {
+	if c.EnabledComponents.Has(name) {
+		return ComponentConfig{
+			LeaderElect:   true,
+			ResourceLock:  c.ResourceLock,
+			LeaseDuration: c.LeaseDuration,
+			RenewDeadline: c.RenewDeadline,
+			RetryPeriod:   c.RetryPeriod,
+		}
+	}
+
+	return defaultComponentConfig()
+}
+
+func defaultConfig() Config {
+	return Config{
+		ResourceLock:      "leases",
+		LeaseDuration:     15 * time.Second,
+		RenewDeadline:     10 * time.Second,
+		RetryPeriod:       2 * time.Second,
+		EnabledComponents: sets.NewString(),
+	}
+}
+
+// ComponentConfig represents the leader election config for a single component.
+type ComponentConfig struct {
+	LeaderElect   bool
+	ResourceLock  string
+	LeaseDuration time.Duration
+	RenewDeadline time.Duration
+	RetryPeriod   time.Duration
+}
+
+func defaultComponentConfig() ComponentConfig {
+	return ComponentConfig{
+		LeaderElect: false,
```

Comment:
This is the default value. I assume you're just trying to make the default very clear?